source/scripting/Checkpoint.cpp

```diff
@@ -0,0 +1,88 @@
+#include "Checkpoint.hpp"
+#include "Native.hpp"
+
+namespace GTA
+{
+    using namespace Math;
+    using namespace System;
+
+    Checkpoint::Checkpoint(CheckpointIcon icon, Math::Vector3 position, Math::Vector3 pointTo, float radius, System::Drawing::Color color)
+        : Checkpoint(icon, position, pointTo, radius, color, 0)
+    {
+    }
+    Checkpoint::Checkpoint(CheckpointIcon icon, Math::Vector3 position, Math::Vector3 pointTo, float radius, System::Drawing::Color color, int reserved) : _icon(icon), _position(position), _pointTo(pointTo), _radius(radius), _color(color), _reserved(reserved)
+    {
+        _handle = Native::Function::Call<int>(Native::Hash::CREATE_CHECKPOINT, static_cast<int>(icon), position.X, position.Y, position.Z, pointTo.X, pointTo.Y, pointTo.Z, radius, static_cast<int>(color.R), static_cast<int>(color.G), static_cast<int>(color.B), static_cast<int>(color.A), reserved);
+    }
+    Checkpoint::~Checkpoint()
+    {
+        Delete();
+    }
+
+    int Checkpoint::Handle::get()
+    {
+        return _handle;
+    }
+    CheckpointIcon Checkpoint::Icon::get()
+    {
+        return _icon;
+    }
+    Vector3 Checkpoint::Position::get()
+    {
+        return _position;
+    }
+    Vector3 Checkpoint::PointTo::get()
+    {
+        return _pointTo;
+    }
+    float Checkpoint::Radius::get()
+    {
+        return _radius;
+    }
+    System::Drawing::Color Checkpoint::Color::get()
+    {
+        return _color;
+    }
+    void Checkpoint::Color::set(System::Drawing::Color color)
+    {
+        Native::Function::Call(Native::Hash::SET_CHECKPOINT_RGBA, Handle, static_cast<int>(color.R), static_cast<int>(color.G), static_cast<int>(color.B), static_cast<int>(color.A));
+        _color = color;
+    }
+
+    void Checkpoint::SetCylinderHeight(float nearHeight, float farHeight, float radius)
+    {
+        Native::Function::Call(Native::Hash::SET_CHECKPOINT_CYLINDER_HEIGHT, Handle, nearHeight, farHeight, radius);
+        _radius = radius;
+    }
+    void Checkpoint::SetIconColor(System::Drawing::Color color)
+    {
+        Native::Function::Call(Native::Hash::_SET_CHECKPOINT_ICON_RGBA, Handle, static_cast<int>(color.R), static_cast<int>(color.G), static_cast<int>(color.B), static_cast<int>(color.A));
+    }
+
+    void Checkpoint::Delete()
+    {
+        Native::Function::Call(Native::Hash::DELETE_CHECKPOINT, Handle);
```

Comment:
Does ```DELETE_CHECKPOINT``` silently ignore an invalid handle? Would have to verify it otherwise, since this method can be called multiple times on a single checkpoint (by the user and then again via the destructor).